crates/engine/tree/src/tree/root.rs

```diff
@@ -243,7 +243,9 @@ fn evm_state_to_hashed_post_state(update: EvmState) -> HashedPostState {
                 })
                 .peekable();
 
-            if destroyed || changed_storage_iter.peek().is_some() {
+            if destroyed {
+                hashed_state.storages.insert(hashed_address, HashedStorage::new(destroyed));
```

Comment:
nit: i think it would be better to be explicit here, given that wiped and destroyed are not the same from `HashedStorage` standpoint, same below where `destroyed == true`
```suggestion
                hashed_state.storages.insert(hashed_address, HashedStorage::new(true));
```